### DISCUSSION TITLE
Instrumentation of a Core Java Application

### DISCUSSION THREAD

**Discussion by suman2007** [open-telemetry/opentelemetry-java-instrumentation#10893] [Q&A]
Hi,
I have a microservice written in core Java(not Spring boot), which does not have Maven or Gradle builder, it has ANT to build the application. How could I proceed with OTel Java Agent and SDK to enable instrumentation on it.

**Comment by laurit** [comment]
Otel java agent is enabled with `-javaagent` command line argument that is passed to the `java` process it does not depend on your build system in any way. As for using the otel api/sdk you can include it in your application the same way as you include your other dependencies.

**Reply by suman2007** [reply]
Thanks for the update, I need the jar for API/SDK to put it under the lib directory. Could you suggest from where will I get the jar file.

**Reply by suman2007** [reply]
**Getting below error while compiling java code**
error: package io.opentelemetry.api.trace does not exist

I have imported import io.opentelemetry.api.trace.Span;
Also placed opentelemetry-api-1.32.0.jar under lib/ directory

[javac] test.java:38: error: package io.opentelemetry.api.trace does not exist
    [javac] import io.opentelemetry.api.trace.Span;
    [javac]                                  ^
    [javac] 1 error

**Reply by laurit** [reply]
> Could you suggest from where will I get the jar file.

You can download it from https://central.sonatype.com/artifact/io.opentelemetry/opentelemetry-api keep in mind that you'll also have to download the dependencies.

> Getting below error while compiling java code
error: package io.opentelemetry.api.trace does not exist

`opentelemetry-api ` jar contains that package.

**Reply by suman2007** [reply]
Thanks
